### ISSUE TITLE
Multi Component Example

### ISSUE THREAD

**Issue by alee792** [vugu/vugu#30]
**Is your feature request related to a problem? Please describe.**
First off, love the project, one of the better Vue/WASM ones out there.
It's a bit difficult to structure a multi-component project with separate Go files outside of the script tag. As someone experienced in Go, but not front end, here are a few remaining questions:

1. Do components need to live in the same dir as the root comp? I couldn't find a pattern where I could have multiple directories or packages.
2. It seems like non-tag comps need to be `// +build ignore` and `package main`. Changing either results in name space collisions. This issue likely goes away if there's a good solution for 1.
3. Are components only allowed to import one and only one component? The documentation seems to suggest this, but I wanted to make sure. If true, this means that components will have to be rolled up via composition at each step, correct?

**Describe the solution you'd like**
A more advanced example incorporating multiple components written outside of script tags. My slapshod attempt to combine the tutorial comps can be found at https://github.com/alee792/vugu-example. 

**Describe alternatives you've considered**
Trial and error 😺

**Comment by bradleypeabody** [comment]
Thanks, I'll do my best to answer:

1. Using components from multiple packages should work, but it's not well tested.  The main reason is there is still some refactor to be done specifically regarding how nested components interact with each other and it will break some things (sorry, experimental!), but once that is done (a week or two?) there will definitely be more documentation.  For the moment, you should be able to: A) Make a separate package with a .vugu file in, B) use [vugugen](https://github.com/vugu/vugu/tree/master/vugugen) to generate your .go file, and C) import that package from your main package (e.g. `import _ "other/package/with/comp"` - that should cause the component in the other package to be available from (main package dir)/root.vugu.

2. .vugu files do not affect the Go build process in any way; the same package naming rules apply - all the files have to have the same package name (unless they are ignored via build tag).  If you change it to something other than "main", the vugu code generator will use that package name, but then your main() func needs to live some place else.  Other than package names, or if you're declaring a "Root" or "RootData" struct in two different places, I don't know what other conflict there could be.

3. I'm not sure what you mean by "import one and only one component" - if you mean each .vugu file corresponds to exactly one component, yes that is correct.  And the idea is you use the technique [here](https://www.vugu.org/doc/components) to include one or more components inside another.

Hopefully that helps for now.

Making it so multiple components work well together and documenting it well is definitely high priority for the project.  It should smooth out once #28 is done.

**Comment by bradleypeabody** [comment]
I'm going to go ahead and close this.  If still concerns or questions please have a look at https://github.com/vugu/vugu/wiki/Refactor-Notes---Sep-2019 for the latest developments on this and file a new issue as needed.
